ValidationLibrary/Rules/HasNotManyStaleBranchesRule.cs

```diff
@@ -0,0 +1,61 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Octokit;
+
+namespace ValidationLibrary.Rules
+{
+    /// <summary>
+    /// This rule checks that repository does not have too many stale branches.
+    /// </summary>
+    public class HasNotManyStaleBranchesRule : IValidationRule
+    {
+        public string RuleName => "Stale branches";
+
+        private readonly ILogger<HasNotManyStaleBranchesRule> _logger;
+
+        public HasNotManyStaleBranchesRule(ILogger<HasNotManyStaleBranchesRule> logger)
+        {
+            _logger = logger;
+        }
+
+        public Task Init(IGitHubClient ghClient)
+        {
+            _logger.LogInformation("Rule {ruleClass} / {ruleName}, Initialized", nameof(HasNotManyStaleBranchesRule), RuleName);
+            return Task.FromResult(0);
+        }
+
+        public async Task<ValidationResult> IsValid(IGitHubClient client, Repository gitHubRepository)
+        {
+            _logger.LogTrace("Rule {ruleClass} / {ruleName}, Validating repository {repositoryName}", nameof(HasNotManyStaleBranchesRule), RuleName, gitHubRepository.FullName);
+
+            var branches = await client.Repository.Branch.GetAll(gitHubRepository.FullName?.Split('/')[0] ?? "", gitHubRepository.Name);
+
+            var staleCommitsMap = new Dictionary<string, bool>();
+            var now = DateTimeOffset.Now;
+            var staleCount = 0;
+
+            foreach (var branch in branches)
+            {
+                if (!staleCommitsMap.ContainsKey(branch.Commit.Sha))
+                {
+                    var commit = await client.Repository.Commit.Get(gitHubRepository.Id, branch.Commit.Sha);
+                    staleCommitsMap[branch.Commit.Sha] = (now - commit.Commit.Author.Date) > TimeSpan.FromDays(90);
```

Comment:
You could just calculate `var staleThreshold = DateTimeOffset.UtcNow - TimeSpan.FromDays(90)` once and use it to make this comparison a bit clearer perhaps?